text/3366-diagnostic-attribute-namespace.md

```diff
@@ -0,0 +1,234 @@
+- Feature Name: The `#[diagnostic]` attribute namespace
+- Start Date: 2023-01-06
+- RFC PR: [rust-lang/rfcs#0000](https://github.com/rust-lang/rfcs/pull/0000)
+- Rust Issue: [rust-lang/rust#0000](https://github.com/rust-lang/rust/issues/0000)
+
+
+# Summary
+[summary]: #summary
+
+This RFC proposed to add a stable `#[diagnostic]` attribute namespace, which contains attributes to influence error messages emitted by the compiler. In addition it proposed to add a `#[diagnostic::on_unimplemented]` attribute to influence error messages emitted by unsatisfied traits bounds.
```

Comment:
```suggestion
This RFC proposed to add a stable `#[diagnostic]` attribute namespace, which contains attributes to influence error messages emitted by the compiler. Attributes in this namespace are versioned and crates that use the `#[diagnostic]` attributes must declare a version at the root level (e.g., `#[diagnostic::v1]`). In addition it proposes to add a `#[diagnostic::on_unimplemented]` attribute to influence error messages emitted by unsatisfied traits bounds.
```

Reply:
Good catch, that's something I've missed after updating the RFC to incorporate a version mechanism. Should I push an update now or should I wait for the [proposed design meeting](https://hackmd.io/e2fvDhvFS_-muKmxOezX_A#%E2%80%9CThe-diagnostic-attribute-namespace%E2%80%9D-rfcs3368) to happen?